components/TextFields/src/ContainedInputView/MDCOutlinedTextField.m

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCOutlinedTextField.h"
+
+#import <Foundation/Foundation.h>
+
+#import "private/MDCBaseTextField+MDCTextControl.h"
+#import "private/MDCTextControl.h"
+#import "private/MDCTextControlStyleOutlined.h"
+
+@interface MDCOutlinedTextField ()
+@end
+
+@implementation MDCOutlinedTextField
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCOutlinedTextFieldInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonMDCOutlinedTextFieldInit];
+  }
+  return self;
+}
+
+- (void)commonMDCOutlinedTextFieldInit {
+  self.containerStyle = [[MDCTextControlStyleOutlined alloc] init];
+  self.borderStyle = UITextBorderStyleNone;
+}
+
+#pragma mark Stateful Color APIs
+
+- (void)setOutlineColor:(nonnull UIColor *)outlineColor forState:(MDCTextControlState)state {
+  [self.outlinedStyle setOutlineColor:outlineColor forState:state];
+  [self setNeedsLayout];
```

Comment:
The layout pass is when the actual color gets applied. I added the setNeedsLayout call because I encountered a bug where this method wasn't doing anything until something else triggered a layout pass.

Reply:
+1 I would have expected a setNeedsDisplay() for a color change.

https://developer.apple.com/documentation/uikit/uiview/1622437-setneedsdisplay?language=objc

Reply:
Color application is tied to the layout cycle. I could be wrong but I don't think `-setNeedsDisplay` is right for this sittuation.